gourmet/reccard.py

```diff
@@ -52,91 +52,91 @@ def __init__ (self, refid, title):
         self.item = title
         self.amount = 1
 
-# OVERARCHING RECIPE CARD CLASS - PROVIDES GLUE BETWEEN EDITING AND DISPLAY
 
-class RecCard (object):
-
-    def __init__ (self, rg=None, recipe=None, manual_show=False):
-        if rg is None:
-            from .GourmetRecipeManager import get_application
-            rg = get_application()
-        self.rg = rg
+class RecCard:
+    """Overarching recipe card class.
+    Provides glue between editing and display.
+    """
+    def __init__(self,
+                 rec_gui: 'RecGui',
+                 recipe: Optional['RowProxy'] = None,
+                 manual_show: bool = False):
+        self.rec_gui = rec_gui
         self.rec_editor: Optional[RecEditor] = None
-        self.conf = []
-        self.new = False
-        if not recipe:
-            recipe = self.rg.rd.new_rec()
-            self.new = True
-        self.current_rec = recipe
+        self.rec_display: Optional[RecCardDisplay] = None
+        self.conf: List = []
+        self.new: bool = True if recipe is None else False
+        self.current_rec: 'RowProxy' = recipe if recipe else self.rec_gui.rd.new_rec()
```

Comment:
First of all, good work! Also on the `set_` and `get_` translations.

Second: Are all these attributes to be "written" by other methods? Some of these look like they should be hidden via double-underscores, others are functional or read-only and could be properties. Do you have any deeper insight?

If we can slim the interface down even further, it would be great.

Reply:
I was not keen to tackle that, but doing so revealed that `RecCard.conf` is a leftover from a previous architecture, and can be removed!